pennylane/ops/qubit/matrix_ops.py

```diff
@@ -236,10 +247,17 @@ def has_decomposition(self) -> bool:
 
     def adjoint(self) -> "QubitUnitary":
         U = self.matrix()
+        if isinstance(U, csr_matrix):
+            adjoint_sp_mat = U.conjugate().transpose()
+            # Note: it is necessary to explicitly cast back to csr, or it will be come csc
+            return QubitUnitary(csr_matrix(adjoint_sp_mat), wires=self.wires)
         return QubitUnitary(qml.math.moveaxis(qml.math.conj(U), -2, -1), wires=self.wires)
```

Comment:
If we are doing everything in CSR format, let's convert it as soon as possible.
```suggestion
    def adjoint(self) -> "QubitUnitary":
        U = self.matrix()
        if isinstance(U, csr_matrix):
            adjoint_sp_mat = csr_matrix(U.conjugate().transpose())
            # Note: it is necessary to explicitly cast back to csr, or it will become csc
            return QubitUnitary(adjoint_sp_mat, wires=self.wires)
        return QubitUnitary(qml.math.moveaxis(qml.math.conj(U), -2, -1), wires=self.wires)
```

Reply:
Resolved at https://github.com/PennyLaneAI/pennylane/pull/6889/commits/a54a8f5051d3682ed6d174c18ea1cbecdde1faea